### PR TITLE
Features/746 print0 print toggle

### PR DESCRIPTION
**Pull request by coquelin77**
## Description

Adds a toggle function to switch between local printing and global printing for easier usability and debugging.

Issue/s resolved: #746 

## Changes proposed:
- added local and global printing options
- added a print0 function to both the heat namespace and the builtins

## Type of change
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - this would only be a breaking change if another function relies on global printing and that is not set explicitly 

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
yes -> the built in print statement will change depending on which mode is selected

### CODE REVIEW

**Comment by coquelin77** [comment]
this might fail on coverage but im not sure how to test `print0`. any ideas for how to test this @Markus-Goetz ?

**Comment by coquelin77** [comment]
rerun tests

File: heat/core/tests/test_printing.py [hunk 1]
```diff
@@ -15,7 +15,44 @@ def tearDown(self):
         # reset the default device
         ht.use_device(self.device)
 
+    def test_local_printing(self):
+        ht.global_printing()
```
Comment (mtar) [inline heat/core/tests/test_printing.py:19]: put in setUp() and delete in all tests

**Review by mtar — CHANGES_REQUESTED** [review]
What do you think about printing the local metadata too? Similar to DNDarray. `LNDarray([], dtype, device, split)` or something like that. It can be useful for debugging.

**Review by mtar — APPROVED** [review]

**Pull request merged**